Library/Homebrew/extend/os/mac/formula.rb

```diff
@@ -1,6 +1,12 @@
 # frozen_string_literal: true
 
 class Formula
+  undef on_macos
+
+  def on_macos(&_block)
+    yield
+  end
```

Comment:
```suggestion
  def on_macos(&block)
    yield if block_given?
  end
```

Reply:
I think these should also raise an error if no block is given.

Reply:
We did not do that for the on_linux/on_macos block below. Should we be consistent? If yes, should I do this in this PR?

Reply:
> Should we be consistent? If yes, should I do this in this PR?

Yes and yes, please!